File: war3map/UnitControllers/HeroUnit.ts
export class HeroUnit {
    public static HERO_POTM: HeroUnit = new HeroUnit("E000", 6);

    public readonly unitType: number;
    public readonly walkAnimationIndex: number;

    constructor(unitType: string, walkAnimationIndex: number) {
        this.unitType = FourCC(unitType);
        this.walkAnimationIndex = walkAnimationIndex;
    }
}

File: src/war3map/UnitControllers/HeroUnit.lua
local ____exports = {}
____exports.HeroUnit = {}
local HeroUnit = ____exports.HeroUnit
HeroUnit.name = "HeroUnit"
HeroUnit.__index = HeroUnit
HeroUnit.prototype = {}
HeroUnit.prototype.__index = HeroUnit.prototype
HeroUnit.prototype.constructor = HeroUnit
function HeroUnit.new(...)
    local self = setmetatable({}, HeroUnit.prototype)
    self:____constructor(...)
    return self
end
function HeroUnit.prototype.____constructor(self, unitType, walkAnimationIndex)
    self.unitType = FourCC(unitType)
    self.walkAnimationIndex = walkAnimationIndex
end
HeroUnit.HERO_POTM = ____exports.HeroUnit.new("E000", 6)
return ____exports
